---
title: "Phenome-wide Multi-Institutional Multimorbidity Explorer Case Study"
author: Siwei Zhang
date: "`r Sys.Date()`"
output:
  rmdformats::downcute:
  style_switcher: false
  thumbnails: no
  lightbox: yes
  gallery: no
  highlight: tango
  use_bookdown: no
  toc_depth: 6
  fig_caption: yes
  embed_fonts: no
  keep_md: no
  number_sections: yes
---
<!-- <style type="text/css"> -->
<!-- .main-container { -->
<!-- max-width: 3000px; -->
<!-- margin-left: auto; -->
<!-- margin-right: auto; -->
<!-- } -->
<!-- </style> -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
embed_png <- function(image_title, width = "100%") {
  knitr::asis_output(
    glue::glue("<div style=\"text-align: center;\">",
                 "<img src=\"{image_title}\" width={width} />",
               "</div>")
  )
}
```

# Introduction

- The JAK2V617F mutation is a somatic mutation in the Janus kinase 2 (JAK2) gene, located on chromosome 9p. This mutation leads to constitutive activation of the JAK-STAT signaling pathway, which is crucial for regulating cell growth, differentiation, and survival.

- Deep sequencing techniques allow for sensitive and accurate detection of the JAK2V617F mutation in patient samples. The Vanderbilt MEGA Genotype Biobank is a large-scale genetic repository, within this cohort, we applied deep sequencing and have identified a cohort of 789 patients with JAK2V617F mutation, and treat other patients as the JAK2V617F mutation non-carriers.

- It's essential to note that while the presence of the JAK2V617F mutation may predispose individuals to myeloproliferative neoplasms (MPNs), polycythemia vera (PV), or essential thrombocythemia (ET), not all carriers of this mutation will develop clinically evident disease. Individuals with a higher variant allele frequency (VAF) of the JAK2V617F mutation are more likely to exhibit overt MPN phenotypes and may be diagnosed with specific MPN subtypes based on established clinical criteria. However, there are also individuals with lower VAF who may harbor the mutation but do not manifest clinically apparent MPN phenotypes.

# Pre-analysis

- We conducted multimorbidity analyses the same to those outlined in PheMIME, but in two distinct phases. Initially, we performed separate analyses for individuals carrying the JAK2V617F mutation and those without it. For each group, we constructed a multimorbidity network. In each analysis, logistic regressions were executed, adjusting for the patient's age at the last recorded visit, sex, race, and the count of unique phecodes present in the patient's records. For every pair of two phecodes, two conditions were considered: one where phecode A was the outcome and phecode B was the independent variable, and vice versa.

  phecode_a ~ phecode_b+age+sex+age+race+EHRage

  phecode_b ~ phecode_a+age+sex+age+race+EHRage

- We excluded disease pairs that showed significant prevalence among JAK2V617F non-carriers. Subsequently, we identified disease phenotypes enriched in the JAK2V617F carrier multimorbidity network and integrated them into associationSubgraphs.

```{r,echo=FALSE, cache=FALSE, warning=FALSE,message=FALSE}
### associationsubgraph
library(tidyverse)
library(glue)
# install.packages("/home/siwei/associationsubgraphs",repos=NULL, type="source")
library(associationsubgraphs,lib.loc = .libPaths()[1])
library(phewasHelper)
# load("phecode_def.rda")

multimorbidity_network = readRDS("multimorbidity_network.rds")

visualize_subgraph_structure(
  multimorbidity_network[[1]],
  node_info = multimorbidity_network[[2]],
  subgraph_results = multimorbidity_network[[3]],
  trim_subgraph_results = TRUE
)


```

- To assess **structure equivalence** within the similarity network, indicating a high correlation between each pair of phenotypes in the multimorbidity pattern, we performed analogous analyses to those described in PheMIME.

- Disease pairs prevalent among JAK2-V617F non-carriers in the multimorbidity similarity networks were excluded. Subsequently, we identified disease phenotypes enriched in the JAK2V617F carrier multimorbidity similarity network.

```{r,echo=FALSE, cache=FALSE, warning=FALSE,message=FALSE}
similarity_network = readRDS("similarity_network.rds")

visualize_subgraph_structure(
  similarity_network[[1]],
  node_info = similarity_network[[2]],
  subgraph_results = similarity_network[[3]],
  trim_subgraph_results = TRUE
)
```

# Analysis 

## Analysis of **Abnormality of red blood cells** and **pregnancy complications**

- **Abnormality of red blood cells** is in the center surrounded by a bunch of pregnancy complications. Patients with **abnormality of red blood cells** diagnosis is enriched in JAK2 mutation carrier.

- Patients with **abnormality of red blood cells** diagnosis is enriched in JAK2 mutation carrier. Patients with **pregnancy complications** is enriched in JAK2 mutation carrier.

- Shared phenotypes between **pregnancy complications** and **abnormalities of red blood cells** suggest potential commonalities. Notably, the co-occurrence of these conditions with essential hypertension, malaise and fatigue, and joint pain exceeds expected frequencies. These insights could prove valuable for identifying patient subgroups among pregnant women with pregnancy complications and carrying the JAK2V617F mutation.

```{r,echo=FALSE, cache=FALSE, warning=FALSE,message=FALSE,fig.width = 12, fig.height = 12}
library(boot)
library(table1)
library(gtsummary)

jak2_dat = readRDS("jak2_dat.rds") %>%
  mutate(group = factor(group,levels=c("JAK2 V617F carrier","JAK2 V617F non-carrier")))
or_fun <- function(data, variable, by, ...) {
  table(data[[by]], data[[variable]]) %>%
    fisher.test() %>%
    broom::tidy()
}
jak2_dat %>%
  dplyr::select(abnormal_redbloodcell, pregnancy_complications,group) %>%
  tbl_summary(by = group, missing = "no") %>%
  add_difference(test = everything()~ or_fun) %>%
  modify_header(estimate ~ "**Odds Ratio**")

heatmap_dat = readRDS("heatmap_dat.rds")
ggplot(heatmap_dat, aes(a, b, fill= strength)) + 
  geom_tile()+
  labs(x="Phenotypes in subgraph 1", 
              y="Diseases co-occur with abnormaility of red blood cells",
              title="Heatmap",
              subtitle="each cell filled with p-value")+
  geom_text(aes(label = p_val)) + 
  theme(axis.text=element_text(size=10),
        axis.title=element_text(size=10,face="bold"),
        axis.text.x = element_text(angle = 30, hjust=1))

```


# Analysis within pregnant women

- Within pregnant women, JAK2v617f carriers have significantly higher risk of pregnancy complications. 

- Among pregnant women, those carrying the JAK2V617F mutation and experiencing pregnancy complications tend to have a younger age at DNA sampling compared to JAK2V617F carriers without pregnancy complications.

- Among pregnant women, those carrying the JAK2V617F mutation and experiencing pregnancy complications tend to have a lower VAF compared to JAK2V617F carriers without pregnancy complications.

```{r,echo=FALSE, cache=FALSE, warning=FALSE,message=FALSE,fig.width = 8, fig.height = 8}
jak2_dat %>%
  filter(pregnancy=="pregnant") %>%
  dplyr::select(pregnancy_complications,group) %>%
  tbl_summary(by = pregnancy_complications, missing = "no") %>%
  add_difference(test = everything()~ or_fun) %>%
  modify_header(estimate ~ "**Odds Ratio**")

## check the DNA sampling age, age at diagnosis of JAK2v617f mutation
age_dat = jak2_dat %>%
  filter(pregnancy=="pregnant" & group=="JAK2 V617F carrier")
ggplot(age_dat, aes(age, fill= pregnancy_complications)) + 
  geom_density(alpha = 0.3)+
  labs(x="Age at DNA sampling date", 
              y="Density",
              title="Distribution of age at DNA sampling date for JAK2V617F mutation carriers")+
  theme(axis.text=element_text(size=10),
        axis.title=element_text(size=10,face="bold"),
        axis.text.x = element_text(angle = 30, hjust=1))

## predicted VAF for women with pregnancy complications compared to those without pregnancy complications
ggplot(age_dat, aes(x..x.y., fill= pregnancy_complications)) + 
  geom_density(alpha = 0.3)+
  labs(x="predicted VAF", 
              y="Density",
              title="Distribution of VAF for JAK2V617F mutation carriers")+
  theme(axis.text=element_text(size=10),
        axis.title=element_text(size=10,face="bold"),
        axis.text.x = element_text(angle = 30, hjust=1))


```


# PheMIME analysis

- It can indicate associationSubgraph could identify specific subgroups of patients due to different biological mechanisms, which indicates the associationsubgraphs in PheMIME could identify subgroups of patients, and colors help researchers identify those subgraphs co-occurr very often with the selected disease, that is to identify subgroups of patients that highly co-occurr with the selected disease



- Choose a pregnancy complications phenotype


# Conclusion

- PheMIME could help identify patient subtypes
















